chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -157,53 +162,140 @@
         Chef::Log.info("#{name} usurped #{res[1].join(", ")} routes from #{bound_if}") unless res[1].empty?
       end
     end
+
     source = ::Nic.new(bound_if)
     # filter out cluster VIPs that a currently assigned to the interface
     addresses = source.addresses.select { |address| my_addresses.include? address }
     routes = source.routes
-    template "/etc/init.d/ovs-usurp-config-#{name}" do
-      source "ovs-usurp-config.erb"
-      owner "root"
-      group "root"
-      mode "0755"
-      variables(
-        :source => bound_if,
-        :dest => name,
-        :addresses => addresses,
-        :routes => routes
-      )
-      # After the ruby_block "Have #{name} usurp config from #{bound_if}" was
-      # executed for the first time, the physical interface (eth) will not have
-      # any addresses or routes assigned anymore. So we should not recreate the
-      # init script in that case. Neither should it be removed.
-      not_if { addresses.empty? and routes.empty? }
-    end
-    service "ovs-usurp-config-#{name}" do
-      # Don't start it here. It only needs to be executed during boot.
-      action [:nothing]
-      subscribes :enable, resources("template[/etc/init.d/ovs-usurp-config-#{name}]")
+
+    if node[:platform] == "suse" && node[:platform_version].to_f >= 12.0
+      # on SLE12, we create proper network configuration for OVS bridges, see
+      # https://en.opensuse.org/Portal:Wicked/OpenvSwitch
+
+      # We need to create a static ifcfg file so that wicked will know that the
+      # ovs bridge usurps the interface address. This also means that we need
+      # to override the variables we pass to the template for the ifcfg of the
+      # interface.
+      bound_if_resource = resources(:template => "/etc/sysconfig/network/ifcfg-#{bound_if}")
```

Comment:
> Isn't the creation and management of bridges and interfaces a responsibility of barclamp-network?

Yes and no. The issue is that the network cookbook doesn't have information about what is needed here, and that's information internal to the neutron cookbook.

> I was testing that if I check the bridge option for in interface in the node information, and I include the recipe from network, this will update (wrongly) the interfaces.

Hrm, not sure I understand. Can you rephrase or give more details?

> IIUC one of the risks of managing the interfaces here is that barclamp-network can undo this.

When would this happen? As long as the neutron cookbook is being used, then it will not be undone.